### PR TITLE
Image ratio json 1156

### PR DESCRIPTION
**Pull request by igormuba**
When either publishing or editing a post you now update the json metadata with the image ratios.

Issue https://github.com/ecency/ecency-vision/issues/1156

### CODE REVIEW

File: src/common/pages/submit.tsx [hunk 1]
```diff
@@ -609,6 +609,14 @@ class SubmitPage extends BaseComponent<Props, State> {
...
     const [parentPermlink] = tags;
     const jsonMeta = this.buildMetadata();
+    let promissesGetImagesDimensions = [];
+    jsonMeta.image.forEach((element) => {
+      promissesGetImagesDimensions.push(this.getHeightAndWidthFromDataUrl(element));
+    });
+    if (promissesGetImagesDimensions.length > 0) {
+      jsonMeta.image_ratio = await Promise.all(promissesGetImagesDimensions.slice(0, 5));
+    }
+
```
Comment (dkildar) [inline src/common/pages/submit.tsx:619]: Code block below will be better
```
jsonMeta.image_ratio = await Promise.all(
 jsonMeta.image
 .map(element => this.getHeightAndWidthFromDataUrl(element))
 .slice(0, 5)
)
```

File: src/common/pages/submit.tsx [hunk 2]
```diff
@@ -669,14 +677,22 @@ class SubmitPage extends BaseComponent<Props, State> {
...
     );
 
+    let promissesGetImagesDimensions = [];
+    jsonMeta.image.forEach((element) => {
+      promissesGetImagesDimensions.push(this.getHeightAndWidthFromDataUrl(element));
+    });
+    if (promissesGetImagesDimensions.length > 0) {
+      jsonMeta.image_ratio = await Promise.all(promissesGetImagesDimensions.slice(0, 5));
+    }
+
```
Comment (dkildar) [inline src/common/pages/submit.tsx:695]: Same here

**Comment by feruzm** [comment]
@igormuba could you please check review comments made by @dkildar, also fix conflicts?
Also consider renaming field from `image_ratio` to `image_ratios`.

**Review by feruzm — APPROVED** [review]
lgtm

**Comment by noumantahir** [comment]
@feruzm @igormuba 

Is this intentional to store ratios against proxified image urls while in both post body and meta images are referenced by original urls. Asking as I will have to handle proxified urls accordingly as parameters for it vary by device. If it is not by design perhaps it needs to be conveyed.

Screenshot 2023-03-06 at 5 37 31 PM

File: src/common/pages/submit.tsx [hunk 3]
```diff
@@ -610,6 +610,11 @@ class SubmitPage extends BaseComponent<Props, State> {
 
     const [parentPermlink] = tags;
     const jsonMeta = this.buildMetadata();
+    let promissesGetImagesDimensions = [];
+    jsonMeta.image_ratios = await Promise.all(
+      jsonMeta.image.map((element) => this.getHeightAndWidthFromDataUrl(element)).slice(0, 5)
```
[the remark this answers is not in the document]
Comment (feruzm) [reply src/common/pages/submit.tsx:615]: @noumantahir you mean no need to load all images but only slice 5 first and get their sizes?

**Comment by igormuba** [comment]
Update:
https://github.com/ecency/ecency-vision/pull/1260

**Pull request merged**